webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx

```diff
@@ -8,21 +8,17 @@ import LocalizedIcon from 'components/localized_icon';
 import {t} from 'utils/i18n';
 
 type Props = {
-    additionalClassName: string | null;
+    additionalClassName?: string;
 }
 
-export default class WarningIcon extends React.PureComponent<Props> {
-    public static defaultProps: Props = {
-        additionalClassName: null,
-    };
-
-    public render(): JSX.Element {
-        const className = 'fa fa-warning' + (this.props.additionalClassName ? ' ' + this.props.additionalClassName : '');
-        return (
-            <LocalizedIcon
-                className={className}
-                title={{id: t('generic_icons.warning'), defaultMessage: 'Warning Icon'}}
-            />
-        );
-    }
+function WarningIcon(props: Props) {
+    const className = 'fa fa-warning' + (props.additionalClassName ? ' ' + props.additionalClassName : '');
+    return (
+        <LocalizedIcon
+            className={className}
+            title={{id: t('generic_icons.warning'), defaultMessage: 'Warning Icon'}}
```

Comment:
Out of the scope but could you move this out of the component and use `defineMessage`? That should make the life easier to the linter automation. The idea would be to have something like:
```
const iconTitle = {id: 'generic_icons.warning', defaultMessage: 'Warning Icon'};

function WarningIcon(props:Props) {
    ...
    return (
        ...
            title={iconTitle}
```

That way we can get rid of the `t` function, that is something we want to achieve eventually.

Reply:
Just replace it with `defineMessage`. I also confirm it's still working as usual:
<img width="1019" alt="image" src="https://github.com/mattermost/mattermost/assets/37064641/f189f199-4ebf-46f9-8d0d-63792aa08e3b">